packages/dom-ready/README.md

```diff
@@ -32,7 +32,7 @@ domReady( function() {
 
 _Parameters_
 
--   _callback_ `Function`: A function to execute after the DOM is ready.
+-   _callback_ `null`: A function to execute after the DOM is ready.
```

Comment:
Docgen fails to handle the `() => void` type well 😞

Reply:
> Docgen fails to handle the `() => void` type well 😞

Related: #18045

Reply:
It feels like a pretty bad change in documentation, unfortunately. I might like to take a closer look to see if it can be easily improved in `docgen`, since the effort involved with #18045 could be significant and delayed. 

Couple other thoughts:

- Can we leave it as `Function` type for the time being?
- Could we "manually" document this function? Even if it's just removing the docgen "START TOKEN" and "END TOKEN" and manually managing this markdown for the short-term.

Reply:
I tried the TypeScript and Closure compiler style function declarations, one resulted in `null` the other in `undefined.

`Function` doesn't typecheck because the the type doesn't satisfy both the direct invocation and the `EventListener` cases.

I've pushed a change that adds a named typedef (which aren't included in docs 😞) `Callback`, which seems like an acceptable temporary solution.

Thoughts?

Reply:
That seems okay to me as a short-term solution. Could help if we indicate this better to a future developer in how they might be able to remove it, e.g. a "TODO" with a reference to the above issue.